### PR TITLE
`Development`: Make sure to unsubscribe from all observables in the tutorial groups client code

### PR DESCRIPTION
**Pull request by stefanwaldhauser**
#### General.
- [x] This is a minor issue that I tested locally and was confirmed by another developer on a test server.

### Motivation and Context
- Improved tutorial group client code quality by making sure to unsubscribe from all observables (I know angular has built-in unsubscribe for some of them, like route params and HTTP, but I find it better to make sure not to rely on these built-ins). The subscriptions are cleaned up when the component is destroyed, protecting against memory leaks

### Description

- Followed the unsubscribe pattern recommended here: https://stackoverflow.com/questions/38008334/angular-rxjs-when-should-i-unsubscribe-from-subscription

### Steps for Testing
Nothing to test, really.

#### Code Review
- [x] Review 1
- [x] Review 2

### Test Coverage
No Changes

### CODE REVIEW

**Review by evgeny1984 — APPROVED** [review]
The code changes are looking good. Good improvement for avoiding memory leaks.

**Comment by krusche** [comment]
please get more approvals. 1 is not enough

**Review by akesfeden — APPROVED** [review]
Code changes look good and they follow the same pattern as the unsubscribe-on-destroy solutions in other components.

File: src/test/javascript/spec/component/tutorial-groups/tutorial-groups-management/tutorial-group-sessions/tutorial-group-sessions-management/tutorial-group-session-row-buttons.component.spec.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { ComponentFixture, TestBed, fakeAsync } from '@angular/core/testing';
+import { ComponentFixture, TestBed, fakeAsync, tick } from '@angular/core/testing';
```
Comment (JohannesStoehr) [inline src/test/javascript/spec/component/tutorial-groups/tutorial-groups-management/tutorial-group-sessions/tutorial-group-sessions-management/tutorial-group-session-row-buttons.component.spec.ts:1]: Unused import

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]
Code looks good except for this one comment:

File: src/test/javascript/spec/component/tutorial-groups/tutorial-groups-management/tutorial-group-sessions/tutorial-group-sessions-management/tutorial-group-session-row-buttons.component.spec.ts (hunk 1 above)
Comment (stefanwaldhauser) [reply src/test/javascript/spec/component/tutorial-groups/tutorial-groups-management/tutorial-group-sessions/tutorial-group-sessions-management/tutorial-group-session-row-buttons.component.spec.ts]: Fixed in de7d568fe3e1c97706f11fc94e414379e6d1bd47

**Review by JohannesStoehr — APPROVED** [review]
Code looks good to me. Thanks for removing the unused import

**Review by evgeny1984 — APPROVED** [review]
Reapproved after deleting unused import.

**Pull request merged**